pipelines/toast_ground_sim.py

```diff
@@ -1896,129 +470,136 @@ def main():
     # Expand boresight quaternions into detector pointing weights and
     # pixel numbers
 
-    expand_pointing(args, comm, data, mem_counter)
+    expand_pointing(args, comm, data)
+
+    # Purge the pointing if we are NOT going to export the
+    # data to a TIDAS volume
+    if (args.tidas is None) and (args.spt3g is None):
+        for ob in data.obs:
+            tod = ob["tod"]
+            tod.free_radec_quats()
 
     # Prepare auxiliary information for distributed map objects
 
     _, localsm, subnpix = get_submaps(args, comm, data)
 
-    if args.input_pysm_model:
+    if args.pysm_model:
+        focalplanes = [s[3] for s in schedules]
         signalname = simulate_sky_signal(
-            args, comm, data, mem_counter, schedules, subnpix, localsm
+            args, comm, data, focalplanes, subnpix, localsm, "signal"
         )
     else:
-        signalname = scan_sky_signal(args, comm, data, mem_counter, localsm, subnpix)
+        signalname = scan_sky_signal(args, comm, data, localsm, subnpix, "signal")
 
     # Set up objects to take copies of the TOD at appropriate times
 
     totalname, totalname_freq = setup_sigcopy(args)
 
     # Loop over Monte Carlos
 
-    firstmc = int(args.MC_start)
-    nmc = int(args.MC_count)
+    firstmc = args.MC_start
+    nmc = args.MC_count
 
     freqs = [float(freq) for freq in args.freq.split(",")]
     nfreq = len(freqs)
 
     for mc in range(firstmc, firstmc + nmc):
 
-        simulate_atmosphere(args, comm, data, mc, mem_counter, totalname)
+        simulate_atmosphere(args, comm, data, mc, totalname)
 
         # Loop over frequencies with identical focal planes and identical
         # atmospheric noise.
 
         for ifreq, freq in enumerate(freqs):
 
-            if rank == 0:
+            if comm.world_rank == 0:
                 log.info(
                     "Processing frequency {}GHz {} / {}, MC = {}".format(
                         freq, ifreq + 1, nfreq, mc
                     )
                 )
 
-            copy_atmosphere(args, comm, data, mem_counter, totalname, totalname_freq)
+            # Make a copy of the atmosphere so we can scramble the gains and apply
+            # frequency-dependent scaling.
+            copy_signal(args, comm, data, totalname, totalname_freq)
 
-            scale_atmosphere_by_frequency(args, comm, data, freq, totalname_freq, mc)
+            scale_atmosphere_by_frequency(args, comm, data, freq, mc, totalname_freq)
 
             update_atmospheric_noise_weights(args, comm, data, freq, mc)
 
-            add_sky_signal(data, totalname_freq, signalname)
+            # Add previously simulated sky signal to the atmospheric noise.
+
+            add_signal(args, comm, data, totalname_freq, signalname, purge=(nmc == 1))
```

Comment:
rename `nmc` to easier to understand name?

Reply:
`nmc` -> `nsimu`